arcgis-ios-sdk-samples/Geometry/List transformations/ListTransformationsViewController.swift

```diff
@@ -0,0 +1,178 @@
+            self?.mapView.setViewpoint(AGSViewpoint(center: originalGeometry, scale: 5000), duration: 2.0, completion: nil)
+            
+            // set the url for our projection engine data;
+            self?.setPEDataURL()
+        })
+    }
+
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        // Dispose of any resources that can be recreated.
+    }
+    
+    // add a graphic with the given geometry, color and style to the graphics overlay
+    func addGraphic(_ geometry: AGSGeometry, color: UIColor, style: AGSSimpleMarkerSymbolStyle) {
+        let sms = AGSSimpleMarkerSymbol(style: style, color: color, size: 15.0)
+        graphicsOverlay.graphics.add(AGSGraphic(geometry: geometry, symbol: sms, attributes: nil))
+    }
+    
+    // set up our datumTransformations array
+    func setupTransformsList() {
+        guard let map = mapView.map,
+            let inputSR = originalGeometry.spatialReference,
+            let outputSR = map.spatialReference else { return }
+        
+        // if orderByMapExtent is on, use the map extent when retrieving the transformations
+        if orderByMapExtent.isOn {
+            datatumTransformations = AGSTransformationCatalog.transformationsBySuitability(withInputSpatialReference: inputSR, outputSpatialReference: outputSR, areaOfInterest: mapView.visibleArea?.extent)
+        }
+        else {
+            datatumTransformations = AGSTransformationCatalog.transformationsBySuitability(withInputSpatialReference: inputSR, outputSpatialReference: outputSR)
+        }
+        
+        defaultTransformation = AGSTransformationCatalog.transformation(forInputSpatialReference: inputSR, outputSpatialReference: outputSR)
+        tableView.reloadData()
+    }
+    
+    func setPEDataURL() {
+        // find the PEDataRuntime folder from the documents directory
+        // added using iTunes
+        let path = NSSearchPathForDirectoriesInDomains(FileManager.SearchPathDirectory.documentDirectory, FileManager.SearchPathDomainMask.userDomainMask, true)
+        let subpaths = FileManager.default.subpaths(atPath: path[0])!
+        
+        // search for PEDataRuntime matches
+        let predicate = NSPredicate(format: "SELF MATCHES %@", "PEDataRuntime")
+        let peDataRuntimePaths = subpaths.filter({ (objc) -> Bool in
+            return predicate.evaluate(with: objc)
+        })
+
+        // use first matching path as path to PE data
+        if let documentPEDataRuntime = peDataRuntimePaths.first {
+            // found "PEDataRuntime" folder, create full url
+            let peDataURL = URL(fileURLWithPath: path[0]).appendingPathComponent(documentPEDataRuntime)
+            try? AGSTransformationCatalog.setProjectionEngineDirectory(peDataURL)
+        }
+        
+        setupTransformsList()
+    }
+
+    @IBAction func oderByMapExtentValueChanged(_ sender: Any) {
+        setupTransformsList()
+    }
+
+    //MARK: - TableView data source
```

Comment:
Please implement `UITableViewDataSource.numberOfSections(in:)`. I realize the default is `1`, which is correct in this case, but it is highly unusual to not implement that data source method.

Reply:
Done.